src/Migrator/index.ts

```diff
@@ -583,9 +583,15 @@ export class Migrator extends EventEmitter implements MigratorContract {
       }
     } catch (error) {
       this.error = error
+      throw error
     }
 
-    await this.shutdown()
+    try {
+      await this.shutdown()
```

Comment:
If the `runUp` or `runDown` method raises exception, then the `shutdown` method won't be called after this change

Reply:
True, but I think that the only errors thrown are before any database actions have been run anyway? I.e. if getting the lock fails, then there is no reason to run the shutdown function to release the lock (which is the issue I'm having).